File: app/app/src/main/java/me/williamhester/reddit/ui/ContentClickCallbacks.java
package me.williamhester.reddit.ui;

import android.os.Parcelable;

import me.williamhester.reddit.models.Link;

/** Callback methods for content links being clicked */
public interface ContentClickCallbacks {
  /** Navigates to a given link. */
  void onLinkClicked(Link link);

  /** Navigates to a Parcelable (typically a Submission or User). */
  void navigateTo(Parcelable parcelable);
}

File: app/app/src/main/java/me/williamhester/reddit/ui/ContentClickCallbacks.kt
package me.williamhester.reddit.ui

import android.os.Parcelable

import me.williamhester.reddit.models.Link

/** Callback methods for content links being clicked  */
interface ContentClickCallbacks {
  /** Navigates to a given link.  */
  fun onLinkClicked(link: Link)

  /** Navigates to a Parcelable (typically a Submission or User).  */
  fun navigateTo(parcelable: Parcelable)
}
